big_tests/tests/muc_SUITE.erl

```diff
@@ -3442,6 +3440,42 @@ configure_anonymous(Config1) ->
     end),
     destroy_room(Config).
 
+cancel_iq_sent_to_locked_room_destroys_it(Config) ->
+    escalus:fresh_story(Config, [{alice, 1}], fun(Alice) ->
+        Room = fresh_room_name(),
+        Nick = <<"the-owner">>,
+
+        PresenceResp = escalus:send_and_wait(Alice, stanza_muc_enter_room(Room, Nick)),
+        escalus:wait_for_stanza(Alice),
+        ?assert(is_presence_with_affiliation(PresenceResp, <<"owner">>)),
+        ?assert(is_presence_with_role(PresenceResp, <<"moderator">>)),
+
+        PresenceLeave = escalus:send_and_wait(Alice, stanza_cancel(Room)),
+        escalus:assert(is_presence_with_type, [<<"unavailable">>], PresenceLeave),
+        escalus:assert(is_stanza_from, [room_address(Room, Nick)], PresenceLeave),
+        escalus:assert(is_iq_result, escalus:wait_for_stanza(Alice))
+    end).
+
+cancel_iq_sent_to_unlocked_room_has_no_effect(Config) ->
+    escalus:fresh_story(Config, [{alice, 1}], fun(Alice) ->
+        Room = fresh_room_name(),
+        Nick = <<"the-owner">>,
+
+        PresenceResp = escalus:send_and_wait(Alice, stanza_muc_enter_room(Room, Nick)),
+        escalus:wait_for_stanza(Alice),
+        ?assert(is_presence_with_affiliation(PresenceResp, <<"owner">>)),
+        ?assert(is_presence_with_role(PresenceResp, <<"moderator">>)),
+
+        InstantRoomIq = stanza_instant_room(Room),
+        InstantRoomIqResp = escalus:send_and_wait(Alice, InstantRoomIq),
+        escalus:assert(is_iq_result, [InstantRoomIq], InstantRoomIqResp),
+        RoomsIqResp = escalus:send_and_wait(Alice, stanza_room_list_request(<<"id">>, undefined)),
+        has_room(room_address(Room), RoomsIqResp),
+
+        CancelIqResp = escalus:send_and_wait(Alice, stanza_cancel(Room)),
+        escalus:assert(is_iq_result, [CancelIqResp], CancelIqResp)
```

Comment:
It's optional.

Still functional means that it's still in normal state and we can join, leave and send messages (+ more checks if needed).
The check can go into a helper function `assert_room_is_functional`.